mits/models.py

```diff
@@ -3,34 +3,64 @@
 
 @Session.model_mixin
 class SessionMixin:
-    def logged_in_mits_team(self):
+    def log_in_as_mits_team(self, team_id, redirect_to='../mits_applications/index'):
         try:
-            return self.mits_team(cherrypy.session['mits_team_id'])
+            team = self.mits_team(team_id)
+            duplicate_teams = []
+            while team.duplicate_of:
+                duplicate_teams.append(team.id)
+                team = self.mits_team(team.duplicate_of)
+                assert team.id not in duplicate_teams, 'circular reference in duplicate_of: {}'.format(duplicate_teams)
         except:
+            log.error('attempt to log into invalid team {}', team_id, exc_info=True)
             raise HTTPRedirect('../mits_applications/login_explanation')
+        else:
+            cherrypy.session['mits_team_id'] = team.id
+            raise HTTPRedirect(redirect_to)
 
-    def mits_teams(self):
+    def logged_in_mits_team(self):
+        try:
+            team = self.mits_team(cherrypy.session['mits_team_id'])
+            assert not team.deleted or team.duplicate_of
+        except:
+            raise HTTPRedirect('../mits_applications/login_explanation')
+        else:
+            if team.duplicate_of:
+                # the currently-logged-in team was deleted, so log back in as the correct team
+                self.log_as_as_mits_team(team.id)
+            else:
+                return team
+
+    def mits_teams(self, include_deleted=False):
+        deleted_filter = [] if include_deleted else [MITSTeam.deleted == False]
         return (self.query(MITSTeam)
-                    .options(joinedload(MITSTeam.applicants),
+                    .filter(*deleted_filter)
+                    .options(joinedload(MITSTeam.applicants).subqueryload(MITSApplicant.attendee),
                              joinedload(MITSTeam.games),
                              joinedload(MITSTeam.schedule),
-                             joinedload(MITSTeam.pictures))
+                             joinedload(MITSTeam.pictures),
+                             joinedload(MITSTeam.documents))
                     .order_by(MITSTeam.name))
 
-    def delete_mits_picture(self, picture):
+    def delete_mits_file(self, model):
```

Comment:
Since we now have two different types of models (pictures and documents) I refactored this method to work on any model with a ``filepath`` field.